### PR TITLE
PET-19 googleOAuth and controler

### PR DESCRIPTION
**Pull request by OmarMahmoud11**

### CODE REVIEW

File: pvsreact/src/pages/Signup.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        navigate('/SignupController', { replace: true, state: { userName, password, email } });
+    }
+    return (
+        <>
+            <div className={styles.signin}>
+                <div className={styles.content}>
+                    <form onSubmit={handleSubmit}>
+                        <Link to="/Signin" className={styles.home}><FaHome></FaHome> Return</Link>
+                        <div className={styles.headup}>Sign up</div>
+                        <input type="text" placeholder="Username" value={userName} onChange={handleUserName} required></input>
```
Comment (karimfathy054) [inline pvsreact/src/pages/Signup.js:43]: if the backend doesn't check for the email format validity , there should be a regex to handle that

**Review by karimfathy054 — CHANGES_REQUESTED** [review]

File: pvsreact/src/pages/Signin.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        console.log(password);
+        navigate('/SigninController', { replace: true, state: { userName, password } });
+    }
+    return (
+        <>
+            <div className={styles.signin}>
+                <div className={styles.content}>
+                    <form onSubmit={handleSubmit}>
+                        <div className={styles.head}>Sign in</div>
+                        <input type="text" placeholder="Username" value={userName} onChange={handleUserName} required></input>
```
[the remark this answers is not in the document]
Comment (OmarMahmoud11) [reply pvsreact/src/pages/Signin.js:29]: There is no email in log in !

File: pvsreact/src/pages/Signup.js (hunk 1 above)
Comment (OmarMahmoud11) [reply pvsreact/src/pages/Signup.js:40]: The email input handle the regex

File: pvsreact/src/Controller/SignupController.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    console.log(encodedEmail);
+
+    const url = `http://localhost:8080/signup?user_name=${encodedUserName}&password=${encodedPassword}&email=${encodedEmail}`;
+    fetch(url, {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ userName, password, email }),
+    }).then((r) => r.json()).then((data) => console.log(data));
```
[the remark this answers is not in the document]
Comment (OmarMahmoud11) [reply pvsreact/src/Controller/SignupController.js:19]: it just to test

File: pvsreact/src/Controller/SigninController.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    const encodedPassword = encodeURIComponent(password);
+
+    const url = `http://localhost:8080/login?user_name=${encodedUserName}&password=${encodedPassword}`;
+    fetch(url, {
+        method: "GET",
+        headers: {
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ userName, password }),
+    }).then((r) => r.json()).then((data) => console.log(data));
```
[the remark this answers is not in the document]
Comment (OmarMahmoud11) [reply pvsreact/src/Controller/SigninController.js:16]: it just to test

File: pvsreact/src/Controller/GoogleOAuthSignupController.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export default function GoogleOAuthSignupController() {
+    const url = `http://localhost:8080/oauthSignUp`;
+    fetch(url, {
+        method: "GET",
+        headers: {
+            "Content-Type": "application/json",
+        }
+    }).then((r) => r.json()).then((data) => console.log(data));
```
[the remark this answers is not in the document]
Comment (OmarMahmoud11) [reply pvsreact/src/Controller/GoogleOAuthSignupController.js:8]: it just to test

File: pvsreact/src/Controller/GoogleOAuthSigninController.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export default function GoogleOAuthSigninController() {
+    const url = `http://localhost:8080/oauthLogin`;
+    fetch(url, {
+        method: "GET",
+        headers: {
+            "Content-Type": "application/json",
+        }
+    }).then((r) => r.json()).then((data) => console.log(data));
```
[the remark this answers is not in the document]
Comment (OmarMahmoud11) [reply pvsreact/src/Controller/GoogleOAuthSigninController.js:8]: it just to test

File: pvsreact/src/pages/GoogleOAuth.js
Comment (OmarMahmoud11) [inline pvsreact/src/pages/GoogleOAuth.js:1]: There is no need for this file so we will delete it

**Pull request merged**